contracts/token/Token.sol

```diff
@@ -385,6 +387,15 @@ contract Token is IToken, AgentRoleUpgradeable, TokenStorage {
         revert RecoveryNotPossible();
     }
 
```

Comment:
to keep compatibility with most DeFi platforms and other external tools we cannot have something so custom, we need to respect ERC20, therefore the function `allowance` should be overriden to return max(uint256) when defaultAllowance is set for that spender address and that the owner didn't opt out of the default allowance (we should provide a function for token holder to manage allowance normally and refuse the default allowance if they deem it preferable)

Reply:
in current implementation the `allowance` would return for example 0 while the transferFrom is going to pass, it is going to create a lot of bugs as DeFi platforms are implemented to ask you to increase allowance if `allowance` returns an amount lower than the amount to transfer